tests/unittests/Foundation/NSFileManagerTests.mm

```diff
@@ -136,4 +161,55 @@
     NSString* currentPath = [manager currentDirectoryPath];
     ASSERT_OBJCNE_MSG(originalPath, currentPath, "Expected change in current directory");
     ASSERT_OBJCEQ_MSG(parentPath, currentPath, "Expected current directory to change to parentPath");
-}
\ No newline at end of file
+}
+
+TEST(NSFileManager, MoveFileViaPath) {
+    unique_fileDeleter srcName(@"NSFileManagerMoveTestFilePath.txt", deleteFile);
+    unique_fileDeleter destName(@"MovedFilePath.txt", deleteFile);
+
+    NSString* content = @"The Quick Brown Fox.";
+    createFileWithContentAndVerify(srcName.get(), content);
+
+    NSString* srcPath = getPathToFile(srcName.get());
+    NSString* destPath = getPathToFile(destName.get());
+
+    NSFileManager* manager = [NSFileManager defaultManager];
+
+    NSError* error = nil;
+    BOOL status = [manager moveItemAtPath:srcPath toPath:destPath error:&error];
+    ASSERT_TRUE(status);
+    EXPECT_EQ(nil, error);
+
+    // Verify file exists.
+    ASSERT_TRUE([manager fileExistsAtPath:destPath]);
+    EXPECT_FALSE([manager fileExistsAtPath:srcPath]);
+
+    // Verify data.
+    ASSERT_OBJCEQ([content dataUsingEncoding:NSUTF8StringEncoding], [NSData dataWithContentsOfFile:destPath]);
+}
+
+TEST(NSFileManager, MoveFileViaURL) {
+    unique_fileDeleter srcName(@"NSFileManagerMoveTestFileURL.txt", deleteFile);
+    unique_fileDeleter destName(@"MovedFileURL.txt", deleteFile);
+
+    NSString* content = @"The Quick Brown Fox.";
+    createFileWithContentAndVerify(srcName.get(), content);
+
+    NSURL* srcURL = [NSURL fileURLWithPath:getPathToFile(srcName.get())];
+    NSURL* destURL = [NSURL fileURLWithPath:getPathToFile(destName.get())];
+
+    NSFileManager* manager = [NSFileManager defaultManager];
+
+    NSError* error = nil;
+    BOOL status = [manager moveItemAtURL:srcURL toURL:destURL error:&error];
+    ASSERT_TRUE(status);
+    EXPECT_EQ(nil, error);
+
+    // Verify file exists.
+    ASSERT_TRUE([manager fileExistsAtPath:[destURL path]]);
+    EXPECT_FALSE([manager fileExistsAtPath:[srcURL path]]);
+
+    // Verify data.
+
+    ASSERT_OBJCEQ([content dataUsingEncoding:NSUTF8StringEncoding], [NSData dataWithContentsOfURL:destURL]);
+}
```

Comment:
either way we end up encoding, but it's a single pass. changed .

---
In reply to: [71443437](https://github.com/Microsoft/WinObjC/pull/669#discussion_r71443437) [](ancestors = 71443437)